File: app/src/main/java/com/saienko/androidthings/barman/AlcoApplication.java
package com.saienko.androidthings.barman;

import android.app.Application;
import com.saienko.androidthings.barman.db.AppDatabase;
import com.squareup.leakcanary.LeakCanary;

/**
 * Created
 * User: Vasiliy Saienko
 * Date: 9/9/17
 * Time: 13:38
 */

public class AlcoApplication extends Application {


    @Override
    public void onCreate() {
        super.onCreate();
        AppDatabase.init(this);
        if (LeakCanary.isInAnalyzerProcess(this)) {
            // This process is dedicated to LeakCanary for heap analysis.
            // You should not init your app in this process.
            return;
        }
        LeakCanary.install(this);
    }
}

File: app/src/main/java/com/saienko/androidthings/barman/AlcoApplication.kt
package com.saienko.androidthings.barman

import android.app.Application
import com.saienko.androidthings.barman.db.DatabaseCreator
import com.tspoon.traceur.Traceur

/**
 * Created
 * User: Vasiliy Saienko
 * Date: 9/9/17
 * Time: 13:38
 */

class AlcoApplication : Application() {


    override fun onCreate() {
        super.onCreate()
        DatabaseCreator.createDb(this)
//        if (LeakCanary.isInAnalyzerProcess(this)) {
//            //            // This process is dedicated to LeakCanary for heap analysis.
//            //            // You should not init your app in this process.
//            return
//        }
//        LeakCanary.install(this)
        Traceur.enableLogging()
    }
}
